lib/nerves_network/resolvconf.ex

```diff
@@ -136,36 +222,133 @@ defmodule Nerves.Network.Resolvconf do
     for ns <- nslist, do: "nameserver #{ns}\n"
   end
   defp nameserver6_text(_), do: ""
 
+  #By default there are 3 nameservers supported by the Linux resolver. A typical configuration (for IPv4/IPv6 dual stack) would look like this:
+  #/etc/resolv.conf:
+  #  nameserver 8.8.8.8
+  #  nameserver 8.8.4.4
+  #  nameserver 2001:4860:4860::8888
+  #  nameserver 2001:4860:4860::8844
+  #
+  # We allow as well to manually using the :static_nameservers entry in the state. Those will be added to the /etc/resolv.conf file
+  # prior to IPv4 and IPv6 nameserver entries in order to take precedence. This is to support 'static' IPv4/IPv6 settings.
+  # The feature can be used to provide a 'static' nameserver entry on the top of ones dynamically configured by either DHCPv4 and/or DHCPv6.
+  @spec write_resolvconf(%{filename: Path.t, ifmap: ifmap | map}) :: :ok
   defp write_resolvconf(state) do
     Logger.debug fn -> "#{__MODULE__}: write_resolvconf state = #{inspect state}" end
 
-    #IPv4 part
     domains     = Enum.map(state.ifmap, &domain_text/1)
-    nameservers = Enum.map(state.ifmap, &nameserver_text/1)
 
-    #IPv6 part
+    #Static Nameservers
+    static_nameservers =  Enum.map(state.ifmap, &static_nameserver_text/1)
+
+    #Common part IPv4/IPv6/Static
+    nameservers_super = Enum.map(state.ifmap, &nameserver_text/1)
+
+    #IPv6 part - must preceed the IPv4 nameservers
     nameservers6 = Enum.map(state.ifmap, &nameserver6_text/1)
 
-    File.write!(state.filename, domains ++ nameservers ++ nameservers6)
+    #We do not need to store nameservers that are in static_nameservers or ipv6_nameservers. nameservers field reports all set-up nameservers and becomes a superposition
+    #After reading the /etc/resolv.conf file - calling Nerves.Network.Resolvconf.settings(ifname) function
+    nameservers = (nameservers_super -- static_nameservers) -- nameservers6
+
+    Logger.debug fn -> "#{__MODULE__}: static_nameservers = #{inspect static_nameservers}" end
+    Logger.debug fn -> "#{__MODULE__}: nameservers        = #{inspect nameservers}" end
+    Logger.debug fn -> "#{__MODULE__}: nameservers6       = #{inspect nameservers6}" end
+
+    File.write!(state.filename, domains ++ static_nameservers ++ nameservers ++ nameservers6)
+  end
+
+
+  @spec entry_to_map(String.t, String.t, ifmap) :: ifmap
+  #domain and search entries are mutually exclusive man (5) resolv.conf. The last entry in the resolv.conf file wins
+  defp entry_to_map("domain", value, _map), do: %{search: value}
+  defp entry_to_map("search", value, _map), do: %{search: value}
+  defp entry_to_map("nameserver", value, map) do
+    nameservers = map[:"nameservers"]
+    case nameservers do
+      nil -> %{nameservers: [value]}
+      # We are doing it slow way to preserve the order of occurrence of nameserver entries - up to 3 such entries are supported
+      _ -> %{nameservers: nameservers ++ [value]}
+    end
+  end
```

Comment:
I tried the map pattern-matching like %{nameservers: nil} and it did not work as expected :/

Reply:
What? How did it fail?